client/src/components/graph/overlays/centroidLabels.js

```diff
@@ -15,27 +16,74 @@ export default
   showLabels: state.centroidLabels?.showLabels,
 }))
 class CentroidLabels extends PureComponent {
+  static watchAsync(props, prevProps) {
+    return !shallowEqual(props.watchProps, prevProps.watchProps);
+  }
+
   constructor(props) {
     super(props);
     this.state = {
-      status: "pending",
-      labels: new Map(),
-      colorAccessor: null,
+      labelSize: 0,
     };
   }
 
-  componentDidMount() {
-    this.updateState(null);
-  }
-
-  componentDidUpdate(prevProps) {
-    this.updateState(prevProps);
+  componentDidUpdate() {
```

Comment:
Since `componentDidUpdate()` is not called for the initial render, will `overlaySetShowing` not be updated?

Reply:
correct, it will not be updated on the initial render, but the initial render has not loaded data, and therefore would always call it with a 'false'.  I can add that to `componentDidMount()`, but it seems like a noop.

Reply:
BTW, this particular pattern seems really odd.  I couldn't come up with a better solution, but the callback can only be executed in the componentDidUpdate/Mount lifecycle, as you can't set state in the render function.